MultipleInstances/src/main/java/com/microsoft/device/display/samples/multipleinstances/MultipleInstancesBaseActivity.kt

```diff
@@ -36,21 +47,25 @@ abstract class MultipleInstancesBaseActivity : AppCompatActivity(), ScreenInfoLi
         return true
     }
 
-    override fun onResume() {
-        super.onResume()
-        ScreenManagerProvider.getScreenManager().addScreenInfoListener(this)
-    }
-
-    override fun onPause() {
-        super.onPause()
-        ScreenManagerProvider.getScreenManager().removeScreenInfoListener(this)
+    private fun initWindowLayoutInfo() {
+        val windowInfoRepository = windowInfoRepository()
+        lifecycleScope.launch(Dispatchers.Main) {
+            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                windowInfoRepository.windowLayoutInfo.collect { info ->
+                    binding.foldableLayout.postDelayed(
+                        { onScreenInfoChanged(info) },
+                        200L
+                    )
+                }
+            }
+        }
     }
 
-    override fun onScreenInfoChanged(screenInfo: ScreenInfo) {
-        first_screen_text.text = getFirstScreenText()
+    private fun onScreenInfoChanged(windowLayoutInfo: WindowLayoutInfo) {
+        findViewById<AppCompatTextView>(R.id.first_screen_text).text = getFirstScreenText()
 
-        if (screenInfo.isDualMode()) {
-            second_screen_text.text = getSecondScreenText()
+        if (windowLayoutInfo.isInDualMode()) {
+            findViewById<AppCompatTextView>(R.id.second_screen_text).text = getSecondScreenText()
```

Comment:
same here (re: viewbinding usage)

Reply:
Same as above: the view is not a direct child of the binding. 
It is added later by the FoldableLayout